pkg/pillar/diskmetrics/usage.go

```diff
@@ -195,19 +195,114 @@ func DirUsage(log *base.LogObject, dir string) (uint64, error) {
 }
 
 // Dom0DiskReservedSize returns reserved space for EVE-OS
-func Dom0DiskReservedSize(log *base.LogObject, globalConfig *types.ConfigItemValueMap, deviceDiskSize uint64) uint64 {
+// We check that the currently used aka dynamic number does not exceed
+// the statically configured percentage and number
+func Dom0DiskReservedSize(log *base.LogObject, globalConfig *types.ConfigItemValueMap, deviceDiskSize uint64, dynamicUsedByDom0 uint64) uint64 {
 	dom0MinDiskUsagePercent := globalConfig.GlobalValueInt(
 		types.Dom0MinDiskUsagePercent)
 	diskReservedForDom0 := uint64(float64(deviceDiskSize) *
 		(float64(dom0MinDiskUsagePercent) * 0.01))
-	maxDom0DiskSize := uint64(globalConfig.GlobalValueInt(
+	staticMaxDom0DiskSize := uint64(globalConfig.GlobalValueInt(
 		types.Dom0DiskUsageMaxBytes))
-	if diskReservedForDom0 > maxDom0DiskSize {
-		log.Tracef("diskSizeReservedForDom0 - diskReservedForDom0 adjusted to "+
-			"maxDom0DiskSize (%d)", maxDom0DiskSize)
-		diskReservedForDom0 = maxDom0DiskSize
+	newlogReserved := uint64(globalConfig.GlobalValueInt(types.LogRemainToSendMBytes))
+	// Always leave space for /persist/newlogd
+	maxDom0DiskSize := newlogReserved
```

Comment:
If we want to add slack for other directories/features (netdump? kernel dumps/stacks, etc) we would need to also have some way to enforce an upper limit on those directories/features.

collect-info.sh clearly has potential issues since it can generate output the same size of what's already present (thus if /persist/newlog is 2Gb that would add that much to the collect-info output give or take some tgz compression).

So I don't have a good answer for those other than the basic slack in reservefordom0.

Reply:
Memory monitor already includes logic to clean the logs after the size of them reaches 100 MB (it removes the oldest then).